nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Discover/DiscoveryWorker.cs

```diff
@@ -93,13 +94,32 @@ public async Task<WorkspaceDiscoveryResult> RunAsync(string repoRootPath, string
             }
 
             // this next line should throw or something
-            projectResults = await RunForDirectoryAsnyc(repoRootPath, workspacePath);
+            projectResults = await RunForDirectoryAsync(repoRootPath, workspacePath);
         }
         else
         {
             _logger.Info($"Workspace path [{workspacePath}] does not exist.");
         }
 
+        //if any projectResults are not successful, return a failed result
+        if (projectResults.Any(p => p.IsSuccess == false))
+        {
+            var failedProjectResults = projectResults.Where(p => p.IsSuccess == false).First();
+
+            var failedResult = new WorkspaceDiscoveryResult
+            {
+                Path = initialWorkspacePath,
+                DotNetToolsJson = dotNetToolsJsonDiscovery,
+                GlobalJson = globalJsonDiscovery,
+                Projects = projectResults.OrderBy(p => p.FilePath).ToImmutableArray(),
+                ErrorType = ErrorType.DependencyFileNotParseable,
```

Comment:
We can't guarantee the failure was due to `DepenencyFileNotParsable` at this point.  I'd do something like this:

``` csharp
var failingProject = projectResults.FirstOrDefault(p => p.ErrorType is not null);
if (failingProject is not null)
{
    var failedResult = new WorkspaceDiscoveryResult
    {
        Path = ..., // same as above
        DotNetToolsJson = null,
        GlobalJson = null,
        Projects = [],
        ErrorType = failingProject.ErrorType,
        ErrorDetails = failingProject.ErrorDetails,
        IsSuccess = false
    }

    return failedResult;
}
```

And set `DotNetToolsJson` and `GlobalJson` to `null`.  If we failed then we can't guarantee anything and I don't want to try to continue the update job.